test/src/test_utils.c

```diff
@@ -72,3 +74,41 @@ ts_test_error_injection(PG_FUNCTION_ARGS)
 	DEBUG_ERROR_INJECTION(text_to_cstring(name));
 	PG_RETURN_VOID();
 }
+
+TS_FUNCTION_INFO_V1(ts_bgw_wait);
+Datum
+ts_bgw_wait(PG_FUNCTION_ARGS)
+{
+	text *datname = PG_GETARG_TEXT_PP(0);
+	/* The timeout is given in seconds, so we compute the number of iterations
+	 * necessary to get a coverage of that time */
+	uint32 iterations = PG_ARGISNULL(1) ? 5 : (PG_GETARG_UINT32(1) + 4) / 5;
+	Oid dboid = get_database_oid(text_to_cstring(datname), false);
+
+	/* This function contains a timeout of 5 seconds, so we iterate a few
+	 * times to make sure that it really has terminated. */
+	int notherbackends;
+	int npreparedxacts;
+	while (iterations-- > 0)
+	{
+		if (!CountOtherDBBackends(dboid, &notherbackends, &npreparedxacts))
+			PG_RETURN_NULL();
+		ereport(NOTICE,
+				(errmsg("source database \"%s\" is being accessed by other users",
+						text_to_cstring(datname)),
+				 errdetail("There are %d other session(s) and %d prepared transaction(s) using the "
+						   "database.",
+						   notherbackends,
+						   npreparedxacts)));
+	}
+	ereport(ERROR,
+			(errcode(ERRCODE_OBJECT_IN_USE),
+			 errmsg("source database \"%s\" is being accessed by other users",
+					text_to_cstring(datname)),
+			 errdetail("There are %d other session(s) and %d prepared transaction(s) using the "
+					   "database.",
+					   notherbackends,
+					   npreparedxacts)));
+	pg_unreachable();
+	PG_RETURN_NULL();
```

Comment:
```suggestion
	pg_unreachable();
```
this is redundant.

Reply:
Got some warnings earlier about this earlier, but it seems to work without the return.